src/Juvix/FrontendContextualise/ModuleOpen/Environment.hs

```diff
@@ -100,3 +159,189 @@ lookupModMap s =
 removeModMap ::
   HasState "modMap" ModuleMap m => Symbol -> m ()
 removeModMap s = Juvix.Library.modify @"modMap" (Map.delete s)
+
+--------------------------------------------------------------------------------
+-- fully resolve module opens
+--------------------------------------------------------------------------------
+
+-- Precondition ∷ old and new context must be in the same context
+-- also terms must be in either the old map or the new map
+-- any place where this is inconsistent will break resolution
+
+-- | @populateModMap@ populates the modMap with all the global opens
+-- in the module
+populateModMap ::
+  WorkingMaps m => m ()
+populateModMap = do
+  old <- get @"old"
+  new <- get @"new"
+  open <- Juvix.Library.ask @"openMap"
```

Comment:
Why is `ask` not in scope unqualified? Import it?

Reply:
it's a conflict with another function in env

Reply:
alright